py/core/utilities.py

```
@@ -7,24 +7,31 @@
 
 from core.bp_metadata_utils.blueprint_meta_data import BlueprintMetaData
 from models.shared.shared_pb2 import Finding
+import logging
 
 
-def verify_config_arguments(args_obj):
+def verify_config_arguments(args_obj) -> bool:
```

I don't know how crucial this validation is, or extendable we want/expect it to get, but here's a pattern for validation function that I've used and liked in the past: 
Have the function return a dict of `{arg: [errors codes]}`

so for example:
```python
return {
    "org_id": ["missing"]  # or maybe "invalid", "too short", etc.
    ...
}
```

And that makes logging the responsibility of the caller, not the validator. It also allows the validator to report on all the issues it finds all at once, instead of short-circuiting on the first issue with the raise exception pattern we had before.

This pattern does make the code more verbose, because then you have to give a human-friendly message to each error code, but I found it to work better in the long run.

Anyways, it's just a thought.

---

@earafat-oak9 ooo I like, I'll update to follow this format. I wanted to add more strict verifications here to throw warnings if certain fields fail to follow the expected format but my simple regex attempt needed some work. Is this something we'd want to add? Right now they could configure their tython project with totally bogus values and as long as they aren't empty it would seem fine until we try the first POST with their project details and it fails. If we were at least looking for simple clues like proj- / env- prefixes and doing length checks on API keys we could fail earlier and give better feedback. In theory it would be nice if we made sure their project configuration details were all valid on running config and we could query the API here but that might be a bit farther out.